Expose deny_new_usb settings

lsfxz opened the pull request:
(Re-)Implement #71

Seems to work; behaviour verified via getprop.
Dynamic mode sets / unsets `security.deny_new_usb` on lock/unlock as
intended, in other modes it remains enabled/disabled.

Current setting is not pre-selected when opening Settings; I currently
lack the knowledge to quickly fix that aspect.

lsfxz committed:
Expose deny_new_usb settings

(Re-)Implement #71

Seems to work; behaviour verified via getprop.
Dynamic mode sets / unsets `security.deny_new_usb` on lock/unlock as
intended, in other modes it remains enabled/disabled.

Current setting is not pre-selected when opening Settings; I currently
lack the knowledge to quickly fix that aspect.

thestinger commented on src/com/android/settings/security/DenyNewUsbPreferenceController.java:
```diff
@@ -0,0 +1,106 @@
+        updateAvailability();
+
+        if (mDenyNewUsb != null) {
+            mDenyNewUsb.setValue(SystemProperties.get(DENY_NEW_USB_PERSIST_PROP, "disabled"));
```
This line should be removed. It should only change it when the setting is changed.

lsfxz replied:
I've removed and tested it without that line: still works, thanks.
Was somewhat of a leftover from the old implementation's code.

lsfxz committed:
remove unnecessary line left over from old code

thestinger commented:
> Current setting is not pre-selected when opening Settings; I currently lack the knowledge to quickly fix that aspect.

Definitely need to get this fixed too.

lsfxz commented:
Will do, it might just take a few days for me to find the spare time to learn and apply what I'll have learned.

lsfxz committed:
Select correct list option when opening preference

The line removed in the previous commit actually was useful, it seems
like I just migrated it to the wrong place in the newly structured
PreferenceController.

Also renamed updateAvailability() to updatePreferenceState(), because
it does more than update the preference's availability at that point.

lsfxz commented:
Turns out the previously removed line was pretty much exactly what was required to selecting the persisted value when showing the preference, only in the wrong place.

Should now be ready, everything seems to work as it should.
